core/activity/src/dao/event.rs

```diff
@@ -99,65 +106,66 @@ impl<'c> EventDao<'c> {
 
     pub async fn get_events(
         &self,
-        identity_id: &str,
+        after_timestamp: DateTime<Utc>,
+        app_session_id: &Option<String>,
         max_events: Option<u32>,
     ) -> Result<Option<Vec<ProviderEvent>>> {
         use schema::activity::dsl;
         use schema::activity_event::dsl as dsl_event;
 
+        let app_session_id = app_session_id.to_owned();
         let limit = match max_events {
-            Some(val) => min(MAX_EVENTS, val),
+            Some(val) => MAX_EVENTS.min(val as i64),
             None => MAX_EVENTS,
         };
 
         log::trace!("get_events: starting db query");
-        let identity_id = identity_id.to_owned();
-
-        do_with_transaction(self.pool, move |conn| {
-            let results: Option<Vec<Event>> = dsl_event::activity_event
+        readonly_transaction(self.pool, move |conn| {
+            let mut query = dsl_event::activity_event
                 .inner_join(schema::activity::table)
-                .filter(dsl_event::identity_id.eq(identity_id))
```

Comment:
is it on purpose to skip identity filtering here? it seems that now every identity can sniff all events, even those not intended for it

Reply:
reverted